src/Bicep.Core.Samples/Files/InvalidVariables_LF/main.diagnostics.bicep

```diff
@@ -321,3 +321,12 @@ var loopExpression = union([for thing in stuff: 4], [for thing in stuff: true])
 //@[1:10) [BCP126 (Error)] Function "batchSize" cannot be used as a variable decorator. |batchSize|
 var batchSizeMakesNoSenseHere = false
 
+
+//KeyVault Secret Reference
+resource kv 'Microsoft.KeyVault/vaults@2019-09-01' existing = {
+  name: 'testkeyvault'
+}
+
+var keyVaultSecretVar = kv.getSecret('mySecret')
```

Comment:
>getSecret [](http://example.com/codeflow?start=27&length=9)

Can you add a case where the function call is in deeper in an expression and also something involved complex objects?

Reply:
Done in arrays and object